src/shipping/views/PriceRatesUpdate/PriceRatesUpdate.tsx

```diff
@@ -174,26 +176,34 @@ export const PriceRatesUpdate: React.FC<PriceRatesUpdateProps> = ({
   const [updateMetadata] = useMetadataUpdate({});
   const [updatePrivateMetadata] = usePrivateMetadataUpdate({});
 
-  const [codesToDelete, setCodesToDelete] = React.useState([]);
-  const [havePostalCodesChanged, setHavePostalCodesChanged] = React.useState(
-    false
-  );
-  const [originalCodes, setOriginalCodes] = React.useState([]);
-  const [inclusionType, setInclusionType] = React.useState(
-    rate?.postalCodeRules[0]?.inclusionType
-  );
+  const [state, dispatch] = React.useReducer(postalCodesReducer, {
+    codesToDelete: [],
+    havePostalCodesChanged: false,
+    inclusionType: rate?.postalCodeRules[0]?.inclusionType,
+    originalCodes: [],
+    postalCodeRules: rate?.postalCodeRules || []
+  });
+
+  if (
```

Comment:
Would gladly make use of useEffect here, but even with [] passed as second arg, either the postalCodeRules was empty or I fell into endless re-render loop.